File: llm_generated_code/Gemini/CPP/Level1/Q13.cpp
#include <iostream>
#include <vector>

using namespace std;

int maxNonAdjacentSum(const vector<int>& arr) {
  int n = arr.size();

  // Handle base cases
  if (n == 0) {
    return 0;
  } else if (n == 1) {
    return arr[0];
  }

  // Calculate maximum sum for the linear array
  vector<int> dp(n);
  dp[0] = arr[0];
  dp[1] = max(arr[0], arr[1]);

  for (int i = 2; i < n; ++i) {
    dp[i] = max(dp[i - 1], arr[i] + dp[i - 2]);
  }

  // Consider the circular nature of the array
  int includeLast = arr[n - 1] + dp[n - 3];
  int excludeLast = dp[n - 1];

  return max(includeLast, excludeLast);
}

int main() {
  vector<int> arr = {5, 5, 10, 100, 10, 5};
  int maxSum = maxNonAdjacentSum(arr);
  cout << "Maximum sum of non-adjacent elements: " << maxSum << endl;

  return 0;
}

File: llm_generated_code/Gemini/Python/Level1/Q13.py
def max_sum_non_adjacent(arr):
  """
  Finds the maximum sum of non-adjacent elements in a circular array.

  Args:
      arr: The input array.

  Returns:
      The maximum sum.
  """

  n = len(arr)

  # Handle base cases
  if n == 1:
    return arr[0]
  elif n == 2:
    return max(arr[0], arr[1])

  # Calculate the maximum sum considering the first and last elements as non-adjacent
  dp = [0] * n
  dp[0] = arr[0]
  dp[1] = max(arr[0], arr[1])
  for i in range(2, n):
    dp[i] = max(dp[i - 1], dp[i - 2] + arr[i])

  # Consider the circular nature: exclude the first or last element
  max_sum1 = dp[n - 1]  # Excluding the last element
  max_sum2 = dp[n - 2] + arr[0]  # Excluding the first element

  return max(max_sum1, max_sum2)

# Example usage
arr = [1, 2, 3, 1]
max_sum = max_sum_non_adjacent(arr)
print("Maximum sum of non-adjacent elements:", max_sum)
